Index: src/Logger/Handler.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/Logger/Handler.php b/src/Logger/Handler.php
--- a/src/Logger/Handler.php	(revision HEAD)
+++ b/src/Logger/Handler.php	(revision Staged)
@@ -4,25 +4,29 @@
 
 namespace Spiral\RoadRunnerBridge\Logger;
 
+use DateTimeInterface;
 use Monolog\Formatter\FormatterInterface;
 use Monolog\Formatter\LineFormatter;
 use Monolog\Handler\AbstractProcessingHandler;
 use Monolog\Handler\HandlerInterface;
 use Monolog\Level;
-use Monolog\Logger;
 use Monolog\LogRecord;
 use RoadRunner\Logger\Logger as RoadRunnerLogger;
 
 final class Handler extends AbstractProcessingHandler
 {
     public const FORMAT = "%message% %context% %extra%\n";
+    private readonly RoadRunnerLogsMode $loggerMode,
 
     public function __construct(
         private readonly RoadRunnerLogger $logger,
         private readonly ?HandlerInterface $fallbackHandler = null,
         string|FormatterInterface $formatter = self::FORMAT,
+        private readonly string $loggerPrefix = '',
+        ?RoadRunnerLogsMode $loggerMode = null,
     ) {
         parent::__construct();
+        $this->loggerMode = $loggerMode ?? RoadRunnerLogsMode::Production;
 
         if (\is_string($formatter)) {
             $formatter = new LineFormatter($formatter);
@@ -42,18 +46,37 @@
 
     protected function write(array|LogRecord $record): void
     {
-        /** @psalm-suppress InvalidArgument */
-        $message = $record['formatted'];
-        \assert(\is_string($message) || $message instanceof \Stringable);
-
-        $level = $record['level'] instanceof Level ? $record['level']->value : $record['level'];
+        if (\is_array($record) && empty($record)) {
+            throw new \InvalidArgumentException('LogRecord should not be empty if is array');
+        }
+        \assert($record['datetime'] instanceof \DateTimeInterface);
 
-        /** @psalm-suppress DeprecatedConstant */
-        match ($level) {
-            Logger::ERROR, Logger::CRITICAL, Logger::ALERT, Logger::EMERGENCY => $this->logger->error($message),
-            Logger::WARNING => $this->logger->warning($message),
-            Logger::INFO, Logger::NOTICE => $this->logger->info($message),
-            Logger::DEBUG => $this->logger->debug($message),
+        $level = $record['level'] instanceof Level ? $record['level'] : Level::tryFrom($record['level']);
+        $level = match ($level) {
+            Level::Error, Level::Critical => 'error',
+            Level::Warning, Level::Alert, Level::Emergency => 'warning',
+            Level::Info, Level::Notice => 'info',
+            Level::Debug => 'debug',
+            null => throw new \LogicException('Unknown log level: ' . $level),
         };
+
+        $ts = $this->loggerMode === RoadRunnerLogsMode::Development
+            ? $record['datetime']->format(DateTimeInterface::RFC3339)
+            : $record['datetime']->format('Uu000');
+
+        $data = [
+                'level' => $this->loggerMode === RoadRunnerLogsMode::Development ? \strtoupper($level) : $level,
+                'ts' => $ts,
+                'logger' => $this->loggerPrefix . $record['channel'],
+                'msg' => $record['message'],
+            ]
+            + ($record['context'] ?? [])
+            + ($record['extra'] ?? []);
+
+        try {
+            $this->logger->log(\json_encode($data, JSON_THROW_ON_ERROR) . PHP_EOL);
+        } catch (\JsonException $e) {
+            $this->logger->error($e->getMessage() . PHP_EOL);
+        }
     }
 }
Index: src/Logger/JsonHandler.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/Logger/JsonHandler.php b/src/Logger/JsonHandler.php
--- a/src/Logger/JsonHandler.php	(revision HEAD)
+++ b/src/Logger/JsonHandler.php	(revision Staged)
@@ -4,6 +4,7 @@
 
 namespace Spiral\RoadRunnerBridge\Logger;
 
+use DateTimeInterface;
 use Monolog\Formatter\FormatterInterface;
 use Monolog\Formatter\LineFormatter;
 use Monolog\Handler\AbstractProcessingHandler;
@@ -50,7 +51,7 @@
         };
 
         $ts = $this->loggerMode === RoadRunnerLogsMode::Development
-            ? $record['datetime']->format('Y-m-d\TH:i:sO')
+            ? $record['datetime']->format(DateTimeInterface::RFC3339)
             : $record['datetime']->format('Uu000');
 
         $data = [
Index: src/Bootloader/LoggerBootloader.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/Bootloader/LoggerBootloader.php b/src/Bootloader/LoggerBootloader.php
--- a/src/Bootloader/LoggerBootloader.php	(revision HEAD)
+++ b/src/Bootloader/LoggerBootloader.php	(revision Staged)
@@ -10,6 +10,7 @@
 use RoadRunner\Logger\Logger;
 use Spiral\Monolog\Bootloader\MonologBootloader;
 use Spiral\RoadRunnerBridge\Logger\Handler;
+use Spiral\RoadRunnerBridge\Logger\RoadRunnerLogsMode;
 use Spiral\RoadRunnerBridge\RoadRunnerMode;
 
 final class LoggerBootloader extends Bootloader
@@ -35,6 +36,8 @@
                     logger: $logger,
                     fallbackHandler: $fallbackHandler,
                     formatter: $env->get('LOGGER_FORMAT', Handler::FORMAT),
+                    loggerPrefix: $env->get('RR_LOGGER_PREFIX', ''),
+                    loggerMode: RoadRunnerLogsMode::tryFrom($env->get('RR_LOGGER_MODE')),
                 );
             },
         ];
